### ISSUE TITLE
unify checks for zipped extensions

### ISSUE THREAD

**Issue by lbergelson** [broadinstitute/gatk#3792]
We currently have code that checks if a file should be zipped based on it's extension all over the place, and it's not consistent.  Some use `.gz` only, some accept `.bgz.` or `.gzip` as well.   We should factor a out a single method in htsjdk and use it everywhere.

**Comment by droazen** [comment]
There already is such a method in htsjdk, isn't there? `AbstractFeatureReader.hasBlockCompressedExtension()`

**Comment by lbergelson** [comment]
@droazen Yes, but it's not consistently used even in htsjdk, and it should probably be moved to IOUtil instead of AbstractFeatureReader.

**Comment by droazen** [comment]
So the method we want exists, we just need to use it consistently and perhaps rename it. Rewording this ticket to reflect that.
